src/js/components/landing/LandingContent.jsx

```diff
@@ -55,12 +55,41 @@ export default class LandingContent extends React.Component {
         this.refs.modal.openModal();
     }
 
+    toggleExpand() {
+        this.setState({
+            expanded: !this.state.expanded
+        })
+    }
+
     render() {
         const affiliations = this.props.session.user.affiliations;
+
+        //Size to set off the limiting of the recent activity
+        let limit=3;
+
         let agencyName = 'Your Agency';
-        if (affiliations && affiliations.length > 0) {
+        if (affiliations && affiliations.length > 0 && this.state.expanded) {
+            agencyName = affiliations.map((a) => a.agency_name).join(', ');
+        }else if(affiliations && affiliations.length > limit && !this.state.expanded){
+            agencyName = affiliations.map((a) => a.agency_name)[0] + "...";
+        }else if(affiliations && affiliations.length < limit && affiliations.length > 0 && !this.state.expanded){
```

Comment:
Also, with your set of checks, what happens if the state is not expanded but the limit is equal to the length? The shorter version covers that issue by not caring what the state is in the second if.